CRM/Contact/Form/Search/Criteria.php

```diff
@@ -564,4 +564,40 @@ public static function CiviCase(&$form) {
     CRM_Case_BAO_Query::buildSearchForm($form);
   }
 
+  /**
+   * @param $form
+   */
+  public static function recurringcontribution(&$form) {
+    $form->add('hidden', 'hidden_recurringcontribution', 1);
+     // Add field to check if payment is made for recurring contribution
+    $recurringPaymentOptions = array(
+        0 => ts(' All recurring contributions'),
+        1 => ts(' Made at least one payment for the recurring contributions'),
+    );
+    $form->addRadio('contribution_recur_payment_made', ts(''), $recurringPaymentOptions, array('allowClear' => TRUE));
+    CRM_Core_Form_Date::buildDateRange($form, 'contribution_recur_start_date', 1, '_low', '_high', ts('From'), FALSE, FALSE, 'birth');
```

Comment:
Does this improve the dates on the existing search fields for recurring contribution dates in advanced search?